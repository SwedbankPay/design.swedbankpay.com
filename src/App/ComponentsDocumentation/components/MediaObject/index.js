import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import MediaObjectComponent from "@components/MediaObject";
import CodeTags from "@components/CodeTags";

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Use the <CodeTags type="secondary" code=".media" /> class on an object to style it as a small container that feature a left- or right-aligned image/icon alongside textual content.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <MediaObjectComponent imgUrl="https://via.placeholder.com/48x48" heading="Bob Corlsan" text="bob.corlsan@swedbankpay.com" />
            <MediaObjectComponent icon="account_circle" heading="Bob Corlsan" text="bob.corlsan@swedbankpay.com" />
        </ComponentPreview>
    </>
);

const Sizes = () => (
    <>
        <h2 id="sizes">Sizes</h2>
        <p>
            Use <CodeTags type="secondary" code=".media-sm" /> or <CodeTags type="secondary" code=".media-lg" /> to modify the size of your media objects.
        </p>
        <h3>Small</h3>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <MediaObjectComponent size="sm" imgUrl="https://via.placeholder.com/40x40" heading="Bob Corlsan" text="bob.corlsan@swedbankpay.com" />
            <MediaObjectComponent size="sm" icon="account_circle" heading="Bob Corlsan" text="bob.corlsan@swedbankpay.com" />
        </ComponentPreview>
        <h3>Large</h3>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <MediaObjectComponent size="lg" imgUrl="https://via.placeholder.com/56x56" heading="Bob Corlsan" text="bob.corlsan@swedbankpay.com" />
            <MediaObjectComponent size="lg" icon="account_circle" heading="Bob Corlsan" text="bob.corlsan@swedbankpay.com" />
        </ComponentPreview>
    </>
);

const MediaPosition = () => (
    <>
        <h2 id="media-position">Media position</h2>
        <p>Use the <CodeTags type="secondary" code=".media-right" /> class to align the media content to the right.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <MediaObjectComponent mediaRight imgUrl="https://via.placeholder.com/48x48" heading="Bob Corlsan" text="bob.corlsan@swedbankpay.com" />
            <MediaObjectComponent mediaRight icon="account_circle" heading="Bob Corlsan" text="bob.corlsan@swedbankpay.com" />
        </ComponentPreview>
    </>
);

const MediaObject = () => (
    <DocContainer docToc>
        <p className="lead">
            Use media objects to showcase basic user information with text and a small image element or an icon.
        </p>
        <Overview />
        <Sizes />
        <MediaPosition />
    </DocContainer>
);

export default MediaObject;

/* For testing */
export { Overview, Sizes, MediaPosition };
