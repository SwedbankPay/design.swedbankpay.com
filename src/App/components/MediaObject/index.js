import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const MediaObject = ({ size, imgUrl, icon, heading, text, textSmall, mediaRight, muted }) => {
    const mediaHeading = () => {
        const headingClasses = classnames(muted ? "text-muted" : null);

        switch (size) {
            case "sm":
                return <h4 className={headingClasses}>{heading}</h4>;
            case "lg":
                return <h2 className={headingClasses}>{heading}</h2>;
            default:
                return <h3 className={headingClasses}>{heading}</h3>;
        }
    };

    const mediaClasses = classnames(
        "media",
        mediaRight ? "media-right" : null,
        size ? `media-${size}` : null
    );

    const iconClasses = classnames(
        "material-icons",
        muted ? "text-muted" : null
    );

    return (
        <div className={mediaClasses}>
            {(imgUrl || icon) ?
                <div className="media-img">{"\n"}
                    {imgUrl ? <> <img src={imgUrl} /> {"\n"} </> : null}
                    {icon ? <> <i className={iconClasses} aria-hidden="true">{icon}</i> {"\n"} </> : null}
                </div>
                : null}
            <div className="media-body">
                {mediaHeading()}
                <p>{textSmall ? <small>{text}</small> : text}</p>
            </div>
        </div>
    );
};

MediaObject.propTypes = {
    size: PropTypes.oneOf(["sm", "lg"]),
    imgUrl: PropTypes.string,
    icon: PropTypes.string,
    heading: PropTypes.string.isRequired,
    text: PropTypes.string,
    textSmall: PropTypes.bool,
    mediaRight: PropTypes.bool,
    muted: PropTypes.bool
};

export default MediaObject;
