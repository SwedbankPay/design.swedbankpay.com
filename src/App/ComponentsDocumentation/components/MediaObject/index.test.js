import React from "react";
import { shallow } from "enzyme";

import MediaObject, { Overview, Sizes, MediaPosition } from "./index";

describe("Documentation: MediaObject", () => {
    it("is defined", () => {
        expect(MediaObject).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<MediaObject />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("Overview", () => {
        it("is defined", () => {
            expect(Overview).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Overview />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Sizes", () => {
        it("is defined", () => {
            expect(Sizes).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<Sizes />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("MediaPosition", () => {
        it("is defined", () => {
            expect(MediaPosition).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<MediaPosition />);

            expect(wrapper).toMatchSnapshot();
        });
    });
});
