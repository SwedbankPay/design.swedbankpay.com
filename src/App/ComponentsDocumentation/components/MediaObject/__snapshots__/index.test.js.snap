// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation: MediaObject MediaPosition renders 1`] = `
<Fragment>
  <h2
    id="media-position"
  >
    Media position
  </h2>
  <p>
    Use the 
    <CodeTags
      code=".media-right"
      type="secondary"
    />
     class to align the media content to the right.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <MediaObject
      heading="Bob Corlsan"
      imgUrl="https://via.placeholder.com/48x48"
      mediaRight={true}
      text="bob.corlsan@swedbankpay.com"
    />
    <MediaObject
      heading="Bob Corlsan"
      icon="account_circle"
      mediaRight={true}
      text="bob.corlsan@swedbankpay.com"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: MediaObject Overview renders 1`] = `
<Fragment>
  <h2
    id="overview"
  >
    Overview
  </h2>
  <p>
    Use the 
    <CodeTags
      code=".media"
      type="secondary"
    />
     class on an object to style it as a small container that feature a left- or right-aligned image/icon alongside textual content.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <MediaObject
      heading="Bob Corlsan"
      imgUrl="https://via.placeholder.com/48x48"
      text="bob.corlsan@swedbankpay.com"
    />
    <MediaObject
      heading="Bob Corlsan"
      icon="account_circle"
      text="bob.corlsan@swedbankpay.com"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: MediaObject Sizes renders 1`] = `
<Fragment>
  <h2
    id="sizes"
  >
    Sizes
  </h2>
  <p>
    Use 
    <CodeTags
      code=".media-sm"
      type="secondary"
    />
     or 
    <CodeTags
      code=".media-lg"
      type="secondary"
    />
     to modify the size of your media objects.
  </p>
  <h3>
    Small
  </h3>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <MediaObject
      heading="Bob Corlsan"
      imgUrl="https://via.placeholder.com/40x40"
      size="sm"
      text="bob.corlsan@swedbankpay.com"
    />
    <MediaObject
      heading="Bob Corlsan"
      icon="account_circle"
      size="sm"
      text="bob.corlsan@swedbankpay.com"
    />
  </ComponentPreview>
  <h3>
    Large
  </h3>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <MediaObject
      heading="Bob Corlsan"
      imgUrl="https://via.placeholder.com/56x56"
      size="lg"
      text="bob.corlsan@swedbankpay.com"
    />
    <MediaObject
      heading="Bob Corlsan"
      icon="account_circle"
      size="lg"
      text="bob.corlsan@swedbankpay.com"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Documentation: MediaObject renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Use media objects to showcase basic user information with text and a small image element or an icon.
  </p>
  <Overview />
  <Sizes />
  <MediaPosition />
</DocContainer>
`;
